packages/contracts/contracts/optimistic-ethereum/OVM/bridge/messaging/OVM_L1CrossDomainMessenger.sol

```diff
@@ -299,4 +319,33 @@ contract OVM_L1CrossDomainMessenger is iOVM_L1CrossDomainMessenger, Abs_BaseCros
             _message
         );
     }
+
+    /**
+     * Generates the correct cross domain calldata for a message.
+     * @param _target Target contract address.
+     * @param _sender Message sender address.
+     * @param _message Message to send to the target.
+     * @param _messageNonce Nonce for the provided message.
+     * @return ABI encoded cross domain calldata.
+     */
+    function _getXDomainCalldata(
```

Comment:
This function is identical between the `OVM_L{1,2}CrossDomainMessenger` contracts, and I think that property will continue to hold true due to the nature of messages touching both contracts.  Thoughts on a library which these both use?

nit: I think we could also take this opportunity to rename this to `encodeXDomainCalldata` if we like, which is a little clearer IMO.